### PR TITLE
Start separating generated files in unique folder instead of inside the src/example tree

### PR DESCRIPTION
**Pull request by andy31415**
#### Problem
ZAP generated files are in-tree and this has undesired sideffects:
  - code review is harder as every change updates different distinct path. If zap updates are mixed with code updates, reviewers have to jump through files to filter out genreated vs non-generated files
  - relative paths are used extensively and include odities (e.g. data_model generated has same file names as example generated making include order important) go easier unnoticed

#### Change overview

- created a 'zzz_generated' folder (for alpha-sorting to go at the end) for all example generated files. Explicitly named folders as 'zap-generated' instead of 'gen' to keep track of them easier in the future (a gen/include is used for build time include generation, would like zap to be explicitly different)
- added 'output folder' support in the regen all script (make all examples go to the zzz folder)
- Migrated examples and other builds to the new structure
- Used `<....>` includes instead of  `"..."`

#### Testing
Verified that things build (except pump/pump-controller which does not build in ToT yet), CI will also validate things.
Checked that zap-regen-all does not change any output.

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

**Comment by jmartinez-silabs** [comment]
What do you think about also moving the .zap file of all examples and the controllers.

Instead of being located in the "xxx-example/xxx-common" that is now almost empty we could maybe regroup them. It would also make easier to go through them in zap tool gui to enable some clusters to all for example.

**Review by jmartinez-silabs — APPROVED** [review]

**Comment by andy31415** [comment]
> What do you think about also moving the .zap file of all examples and the controllers.
> 
> Instead of being located in the "xxx-example/xxx-common" that is now almost empty we could maybe regroup them. It would also make easier to go through them in zap tool gui to enable some clusters to all for example.

All for it. The PR is already quite large and some updates still need to be done. The src/controller/data_model also needs moving, java and python need moving and the regrouping of examples should happen as well.

Expect to gradually update things. Hoping only this starter PR was painful

**Comment by andy31415** [comment]
@chrisdecenzo @jelderton @LuDuda @mspang @saurabhst ?

**Review by tcarmelveilleux — APPROVED** [review]

**Review by mspang — APPROVED** [review]

File: examples/ota-provider-app/linux/BUILD.gn [hunk 1]
```diff
@@ -29,6 +33,8 @@ executable("chip-ota-provider-app") {
   cflags = [ "-Wconversion" ]
 
   output_dir = root_out_dir
+
+  configs = [ ":config" ]
```
Comment (mspang) [inline examples/ota-provider-app/linux/BUILD.gn:37]: What's this needed for?

File: examples/ota-provider-app/linux/BUILD.gn [hunk 2]
```diff
@@ -29,6 +29,8 @@ executable("chip-ota-provider-app") {
   cflags = [ "-Wconversion" ]
 
   output_dir = root_out_dir
+
+  configs = [ "${chip_root}/examples/ota-provider-app:config" ]
```
Comment (mspang) [inline examples/ota-provider-app/linux/BUILD.gn:33]: This should to go on ota-provider-common not the executable

(I really don't understand how this even compiles for you..

oh, actually, I get it.. you didn't change the #includes in BdxOtaSender.cpp and the other files. Please do that)

**Review by mspang — APPROVED** [review]

File: examples/ota-provider-app/linux/BUILD.gn (hunk 2 above)
Comment (andy31415) [reply examples/ota-provider-app/linux/BUILD.gn]: Updated with more fixes after some chat back and forth.

**Pull request merged**